src/cluster/autoscale.go

```diff
@@ -445,3 +449,23 @@ func getImageVersion(clusterID uint, cluster interface{}) map[string]string {
 
 	return selectedImageVersion
 }
+
+func setArchNodeSelector(cluster CommonCluster) map[string]string {
+	k8sVersion, err := getK8sVersion(cluster)
+	if err != nil {
+		log.Error(errors.WrapIfWithDetails(err, "unable to retrieve K8s version of cluster", "clusterID", cluster.GetID()))
```

Comment:
I'd probably return the value and an error from this function. If the value is empty, the caller doesn't need to do anything. If it's not empty, the caller should add the arch label. If it returns an error, the caller should decide how to handle that.